bot/exts/fun/uwu.py

```diff
@@ -0,0 +1,134 @@
+import random
+import re
+from functools import partial
+from typing import Callable
+
+from discord.ext import commands
+from discord.ext.commands import Cog, Context, clean_content
+
+from bot.bot import Bot
+from bot.exts.fun.fun import Fun
+from bot.utils import helpers
+
+
+WORD_REPLACE_HASH = {
+    "small": "smol",
+    "cute": "kawaii~",
+    "fluff": "floof",
+    "love": "luv",
+    "stupid": "baka",
+    "idiot": "baka",
+    "what": "nani",
+    "meow": "nya~",
+    "roar": "rawrr~",
+}
+
+EMOJI_LUT = [
+    "rawr x3",
+    "OwO",
+    "UwU",
+    "o.O",
+    "-.-",
+    ">w<",
+    "σωσ",
+    "òωó",
+    "ʘwʘ",
+    ":3",
+    "XD",
+    "nyaa~~",
+    "mya",
+    ">_<",
+    "rawr",
+    "uwu",
+    "^^",
+    "^^;;",
+]
+
+REGEX_WORD_REPLACE = re.compile(r"(?<![w])[lr](?![w])")
+
+REGEX_PUNCTUATION = re.compile(r"[.!?\r\n\t]")
+
+REGEX_STUTTER = re.compile(r"(\s)([a-zA-Z])")
+SUBSTITUTE_STUTTER = r"\g<1>\g<2>-\g<2>"
```

Comment:
Curious, why isn't this compiled too?

Reply:
The `repl` argument of [`re.sub()`](https://docs.python.org/3/library/re.html#re.sub) requires a string, not a pattern object.

Reply:
https://github.com/python/typeshed/blob/master/stdlib/typing.pyi#L656
The type hint for the `repl` argument is `AnyStr`.

https://github.com/python/cpython/blob/main/Lib/re.py#L277-L281
But the cpython source shows that a compiled pattern is allowed if there are no flags.

Is the compiled pattern okay, or is it required to respect the type hint, and to leave the pattern as a string?

Reply:
Never mind. It can't be compiled ahead of time anyways because the backreference doesn't exist yet.